src/Raven.Client/Documents/AI/AiOperations.cs

```diff
@@ -64,121 +64,41 @@ public AiOperations ForDatabase(string databaseName)
-    /// <param name="prompt">The initial user prompt.</param>
-    /// <param name="parameters">Optional dictionary of chat parameters.</param>
-    /// <returns>The result of the chat.</returns>
-    public ChatResult<TSchema> StartChat<TSchema>(string identifier, string prompt, Dictionary<string, object> parameters = null) where TSchema : new()
-    {
-        return AsyncHelpers.RunSync(() => StartChatAsync<TSchema>(identifier, prompt, parameters));
-    }
-
+    /// <typeparam name="TSchema">The schema type for the chat response.</typeparam>
+    /// <param name="agentName">The name of the AI agent to chat with.</param>
+    /// <param name="parameters">Required chat parameters.</param>
+    public IChatOperations<TSchema> CreateChat<TSchema>(string agent, Dictionary<string, object> parameters = null) where TSchema : new() => new Chat<TSchema>(this, agent, parameters);
+    
     /// <summary>
-    /// Continues a chat using a fluent parameter builder.
-    /// Also update the chat with the new prompt and response.
+    /// Create a new chat with an AI agent using a dictionary of parameters.
     /// </summary>
-    /// <typeparam name="TSchema">The schema type for the chat.</typeparam>
-    /// <param name="chatId">The ID of the existing chat.</param>
-    /// <param name="prompt">The next user prompt in the conversation.</param>
-    /// <returns>The result of the continued chat.</returns>
-    public ChatResult<TSchema> ContinueChat<TSchema>(string chatId, string prompt) where TSchema : new()
-    {
-        return AsyncHelpers.RunSync(() => ContinueChatAsync<TSchema>(chatId, prompt));
-    }
-
-    /// <summary>
-    /// Asynchronously continues a chat using a fluent parameter builder.
-    /// Also update the chat with the new prompt and response.
-    /// </summary>
-    /// <typeparam name="TSchema">The schema type for the chat.</typeparam>
-    /// <param name="chatId">The ID of the existing chat.</param>
-    /// <param name="prompt">The next user prompt in the conversation.</param>
-    /// <returns>The result of the continued chat.</returns>
-    public async Task<ChatResult<TSchema>> ContinueChatAsync<TSchema>(string chatId, string prompt, CancellationToken token = default) where TSchema : new()
-    {
-        return await _executor.SendAsync(new ResumeChatOperation<TSchema>(chatId, userPrompt: prompt), token).ConfigureAwait(false);
-    }
+    /// <typeparam name="TSchema">The schema type for the chat response.</typeparam>
+    /// <param name="agentName">The name of the AI agent to chat with.</param>
+    /// <param name="builder">A builder function to define the required chat parameters.</param>
+    public IChatOperations<TSchema> CreateChat<TSchema>(string agent, Func<AiAgentParametersBuilder, AiAgentParametersBuilder> builder) where TSchema : new() => 
+        CreateChat<TSchema>(agent, builder.Invoke(new AiAgentParametersBuilder()).GetParameters());
```

Comment:
Why do we want this?

Reply:
1. agentName vs agentId vs identifier issue
2. should be Action<IAiAgentParametersBuilder
3. @ayende I prefer that API that passing a dict, especially since parameters are case insensitive as we agreed, so it is easier to validate that via builder